### PR TITLE
[vcpkg_from_git] Fix error if downloads folder does not exist

### PR DESCRIPTION
**Pull request by ras0219-msft**
From https://github.com/microsoft/vcpkg/pull/18705:

> Related issue #18291, #16282, #14424, #18071, #17704, #17517, #16846, #16015, #15911, #15649, #15648, #14974, #13851

This PR actually fixes the root cause.

### CODE REVIEW

**Comment by BillyONeal** [comment]
I don't think this is sufficient because there are similar places all over the tool that expect the downloads directory to already exist, in particular bootstrap and friends. Do we know how we got to this 'downloads doesn't exist' condition in the first place?

**Comment by strega-nil-ms** [comment]
@BillyONeal everywhere else should use `vcpkg_download_distfile`, which already takes care of this.

File: scripts/cmake/vcpkg_from_git.cmake [hunk 1]
```diff
@@ -82,6 +82,7 @@ function(vcpkg_from_git)
     message(STATUS "Fetching ${_vdud_URL}...")
     find_program(GIT NAMES git git.cmd)
     # Note: git init is safe to run multiple times
+    file(MAKE_DIRECTORY ${DOWNLOADS})
```
Comment (strega-nil-ms) [inline scripts/cmake/vcpkg_from_git.cmake:85]:
```suggestion
    file(MAKE_DIRECTORY "${DOWNLOADS}")
```

**Review by strega-nil-ms — APPROVED** [review]

**Comment by BillyONeal** [comment]
> @BillyONeal everywhere else should use `vcpkg_download_distfile`, which already takes care of this.

bootstrap does *not* use `vcpkg_download_distfile`

**Comment by strega-nil-ms** [comment]
@BillyONeal `bootstrap.ps1` does not need `downloads`, while `bootstrap.sh` uses `curl --create-dirs`.

**Review by vicroms — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Closing for rollup #18838

**Pull request closed without merging**